tools/ips.py

```diff
@@ -36,31 +36,36 @@ def generate_hwaddr():
 
 
 def generate_dev(device_name, ip_addr):
-    cmd = 'ip link add ' + device_name + ' type dummy && '
+    cmd = 'sudo ip link add ' + device_name + ' type dummy && '
     cmd += 'ifconfig ' + device_name + ' hw ether ' + generate_hwaddr() + ' && '
     cmd += 'ip addr add ' + ip_addr + ' dev ' + device_name + ' && '
     cmd += 'ip link set ' + device_name + ' up'
+    print(cmd)
     return os.system(cmd)
 
 
 def remove_dev(device_name, ip_addr):
-    return os.system('ip link delete ' + device_name)
+    cmd = 'sudo ip link delete ' + device_name
+    print(cmd)
+    return os.system(cmd)
 
 
 def add_addr_to_existing_dev(device_name, ip_addr):
     if sys.platform.startswith('linux'):
-        cmd = 'ip addr add ' + ip_addr + ' dev ' + device_name + ' && '
+        cmd = 'sudo ip addr add ' + ip_addr + ' dev ' + device_name + ' && '
         cmd += 'ip link set ' + device_name + ' up'
     else:
-        cmd = 'ifconfig ' + device_name + ' alias ' + ip_addr + '/32'
+        cmd = 'sudo ifconfig ' + device_name + ' alias ' + ip_addr + '/32 &'
```

Comment:
cmd = 'sudo ifconfig ' + device_name + ' alias ' + ip_addr + '/32 &'

added & at the end to make script faster